src/components/DropDownButton/index.tsx

```diff
@@ -84,17 +85,18 @@ const Toggle = ({
 	handler,
 	label,
 	joined,
+	compact,
 	...props
 }: InternalDropDownButtonProps & {
 	open: boolean;
 	handler: React.MouseEventHandler<HTMLElement>;
 }) => {
+	const shouldCompact = useBreakpoint(compact || [false]);
 	if (joined) {
-		if (label || props.icon) {
+		if (shouldCompact) {
```

Comment:
Make this `label || shouldShowLabel`, and then have `{shouldShowLabel && <Box .../>}` instead of having two separate conditions with almost the same code.